src/solvers/dg_structured/containers.jl

```diff
@@ -0,0 +1,48 @@
+struct Interface{RealT<:Real, NDIMS}
+  u_left::Array{RealT, NDIMS} # [variables, i, j]
+  u_right::Array{RealT, NDIMS} # [variables, i, j]
+  orientation::Int64
+  surface_flux_values::Array{RealT, NDIMS} # [variables, i, j]
+end
+
+function Interface{RealT, NDIMS}(nvars, nnodes, orientation) where {RealT<:Real, NDIMS}
+  # TODO Is there a more elegant solution for this?
+  u_left = Array{RealT, NDIMS}(undef, nvars, fill(nnodes, NDIMS-1)...)
+  u_right = Array{RealT, NDIMS}(undef, nvars, fill(nnodes, NDIMS-1)...)
+
+  surface_flux_values = Array{RealT, NDIMS}(undef, nvars, fill(nnodes, NDIMS-1)...)
+
+  return Interface{RealT, NDIMS}(u_left, u_right, orientation, surface_flux_values)
+end
+
+
+struct Element{RealT<:Real, NDIMS}
+  node_coordinates::Array{SVector{NDIMS, RealT}, NDIMS}
+  # node_coordinates::Array{RealT, 2}
+  inverse_jacobian::RealT
+  interfaces::Vector{Interface{RealT, NDIMS}} # [orientation]
+end
```

Comment:
```suggestion
# TODO: AD, needs to be adapted to use `RealT` and `uEltype`, cf. https://github.com/trixi-framework/Trixi.jl/pull/461
struct Element{RealT<:Real, NDIMS}
  node_coordinates::Array{SVector{NDIMS, RealT}, NDIMS}
  # node_coordinates::Array{RealT, 2}
  inverse_jacobian::RealT
  interfaces::Vector{Interface{RealT, NDIMS}} # [orientation]
end
```
I'll just comment here but similar things apply to the other containers as well. You don't need to implement changes like the ones mentioned here. I (or someone else) can also do that later. But we should probably leave some comment here.

Reply:
Done.